File: Strings/ReverseString/Solution1.java
package Strings.ReverseString;

// Using the two pointer method
// Time complexity: O(n), we shall traverse through all n elements of the array
// Space complexiy: O(1), we use constant time
public class Solution1 {
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length - 1;
    
        char temp;
        while (left < right) {
            temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }
}

File: Sum/RunningSum/Kotlin/Solution1.kt
class Solution1 {

    // O(n) time and space complexity
    fun runningSum(nums: IntArray): IntArray {
        val sums = IntArray(nums.size)
        
        sums[0] = nums[0]
        
        for (i in 1 until nums.size) {
            sums[i] = nums[i] + sums[i - 1] 
        }
        
        return sums
    }
}